pyproject.toml

```
@@ -26,7 +26,7 @@ classifiers = [
 ]
 requires-python = ">=3.9, <3.13"
 dependencies = [
-  "pybamm>=23.5",
+  "pybamm>23.5",
```

```suggestion
  "pybamm>=23.9",
```

Let's bump with a greater-than-or-equal-to cap so that the resolver does not allow the installation of 23.9rc0, 23.9rcX or other pre-releases in case someone requests them.